feat(@mastra/core): public API return types for streamVNext()

TylerBarnes opened the pull request:
## Description

<!-- Provide a brief description of the changes in this PR -->

## Related Issue(s)

<!-- Link to the issue(s) this PR addresses, using hashtag notation: #123 -->

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Code refactoring
- [ ] Performance improvement
- [ ] Test update

## Checklist

- [ ] I have made corresponding changes to the documentation (if applicable)
- [ ] I have added tests that prove my fix is effective or that my feature works

DanielSLew committed:
type new mastra stream types

TylerBarnes committed:
fix public api return types

TylerBarnes committed:
fix return type for output object

TylerBarnes committed:
temp disable vnext_generatehandler

TylerBarnes committed:
partial + infer z

TylerBarnes committed:
Merge main into fix/stream-vnext-public-types branch

- Updated method names from stream_vnext/generate_vnext to streamVNext/generateVNext
- Applied proper return type fixes to new method names
- Maintained backward compatibility during transition
- Resolved import conflicts for AISDKV5OutputStream

TylerBarnes committed:
fix stream type mismatch vs global WritableStream

TylerBarnes committed:
add tripwire to v5 output

TylerBarnes committed:
add type assertions for now to make public api types correct

TylerBarnes committed:
fix public type so IDE will autocomplete "mastra" | "aisdk" for format key

TylerBarnes committed:
remove commented code

TylerBarnes committed:
Merge remote-tracking branch 'origin/main' into fix/stream-vnext-public-types

TylerBarnes committed:
fix: remove duplicate imports and getter methods to resolve build errors

TylerBarnes committed:
fix: improve type assertions in output-helpers.ts

Addresses https://github.com/mastra-ai/mastra/pull/6876#discussion_r2289453210
Replaced generic 'as any' with proper type assertions for StepResult fields

CalebBarnes commented on packages/core/src/agent/index.ts:
```diff
@@ -3265,19 +3268,19 @@ Message ${msg.threadId && msg.threadId !== threadObject.id ? 'from previous conv
-    >;
+    return fullOutput as unknown as FORMAT extends 'aisdk'
+      ? Awaited<ReturnType<AISDKV5OutputStream<OUTPUT>['getFullOutput']>>
+      : Awaited<ReturnType<MastraModelOutput['getFullOutput']>>;
```
Should we also be passing OUTPUT to `MastraModelOutput`? since that has `object` as well.

CalebBarnes replied:
object in base output is the same as aisdk output. The aisdk output getter just returns the same one.

TylerBarnes committed:
fix: improve type safety in llm-execution.ts

Addresses https://github.com/mastra-ai/mastra/pull/6876#discussion_r2289453211
Addresses https://github.com/mastra-ai/mastra/pull/6876#discussion_r2289453218

- Extract duplicate payload cast to variable for better performance
- Replace non-null assertion with safe fallback for toolName

TylerBarnes committed:
fix: improve type safety across stream types

Addresses multiple PR review comments:
- https://github.com/mastra-ai/mastra/pull/6876#discussion_r2289453208
- https://github.com/mastra-ai/mastra/pull/6876#discussion_r2289453224
- https://github.com/mastra-ai/mastra/pull/6876#discussion_r2289453227

- Export TextDeltaPayload for consistency
- Use proper CoreMessage[] types instead of any[]
- Remove duplicate reason field in StepFinishPayload
- Fix type definitions at source instead of casting

TylerBarnes committed:
Merge main: resolve import conflict for AISDKV5OutputStream

Keep type imports from main branch instead of value imports

TylerBarnes committed:
fix import

TylerBarnes committed:
use actual stream types we use internally, not the incorrect ai sdk ones

TylerBarnes committed:
hook up object type inference for mastra output

TylerBarnes committed:
Merge branch 'main' into fix/stream-vnext-public-types

TylerBarnes committed:
fixlint

TylerBarnes committed:
Merge branch 'fix/stream-vnext-public-types' of https://github.com/mastra-ai/mastra into fix/stream-vnext-public-types

TylerBarnes committed:
temp any

abhiaiyer91 committed:
Merge branch 'main' into fix/stream-vnext-public-types

TylerBarnes commented on client-sdks/client-js/src/resources/agent.ts:
```diff
@@ -803,7 +803,9 @@ export class Agent extends BaseResource {
-      onChunk: async chunk => {
+      // TODO: casting as any here because the stream types were all typed as any before in core.
+      // but this is completely wrong and this fn is probably broken. Remove ":any" and you'll see a bunch of type errors
+      onChunk: async (chunk: any) => {
```
Yeah but since the chunk types were mostly typed as any this file had a bunch of incorrect code that's now revealed by the types. We should follow up in a later pr fixing it because there are hundreds of cases like this that all need to be fixed

abhiaiyer91 committed:
Merge branch 'main' into fix/stream-vnext-public-types

abhiaiyer91 committed:
Fix build

TylerBarnes committed:
hopefully make test more reliable

TylerBarnes committed:
fix v4 test?

TylerBarnes committed:
add missing import

TylerBarnes committed:
fix more

TylerBarnes committed:
delete irrelevant test

TylerBarnes committed:
Merge branch 'main' into fix/stream-vnext-public-types
